### PR TITLE
GH-988: replace old way of managing fields with new formfield class.

### PR DESCRIPTION
**Pull request by ksuhartono97**
Part 1:

- Replaced objects for formfields with FormField class for consistency
- Reworked how prefillable form fields are stored 
- Adjusted UI to work with new data structures

Todo in part 2: implement ID functionality and conditional forms backend design

### CODE REVIEW

**Comment by ksuhartono97** [comment]
Note that this is not backward compatible. We will have to update all our old forms to contain the correct prefill fields.

File: ui/src/components/forms/FormEditor.js [hunk 1]
```diff
@@ -65,20 +66,18 @@ const FormEditor = (props) => {
...
     }
 
-    const dataObject = {
-      fieldName: 'prefill',
-      fieldType: 'prefill',
-      options: chosenFields,
-      fieldDescription: '',
-      required: true,
-    };
+    console.log(chosenFields);
```
Comment (tamarayustian) [inline ui/src/components/forms/FormEditor.js:69]: remove console log

File: ui/src/components/forms/FormEditor.js [hunk 2]
```diff
@@ -65,20 +66,18 @@ const FormEditor = (props) => {
...
 
-    let temp = [...formFields];
-    try {
-      let spliceAmount = temp[0].fieldType === 'prefill' ? 1 : 0;
-      temp.splice(0, spliceAmount, dataObject);
-    } catch (err) {
-      temp.push(dataObject);
+    let prefillableFields = [];
+    for (let field of chosenFields) {
+      console.log(field);
```
Comment (tamarayustian) [inline ui/src/components/forms/FormEditor.js:73]: remove console log

File: ui/src/components/forms/FormField.js
Comment (tamarayustian) [inline ui/src/components/forms/FormField.js:1]: is this file supposed to be here in /components folder too? or maybe we create new folder called /class at the same level? i think its more appropriate in terms of file structure. but just minor thing

File: ui/src/components/forms/FormEditor.js [hunk 3]
```diff
@@ -91,13 +87,23 @@ const FormEditor = (props) => {
     if (data.options) {
       data.options = data.options.split(';');
     }
+
+    let newField = new FormField(
+      '',
```
Comment (albertparedandan) [inline ui/src/components/forms/FormEditor.js:92]: https://www.npmjs.com/package/uuid

use uuid for unique id

File: ui/src/components/forms/FormField.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class FormField {
+  constructor(id, fieldName, fieldType, fieldDescription, required, options) {
+    this.id = id;
```
Comment (albertparedandan) [inline ui/src/components/forms/FormField.js:3]: no need to pass in id i think. just auto generate using uuid upon creation

https://www.npmjs.com/package/uuid

so something like 

```
import { v4 as uuidv4 } from 'uuid';

class FormField {
  constructor(fieldName, fieldType, fieldDescription, required, options) {
    this.id = uuidv4();
    // blablabla
  }
}
```

File: ui/src/components/forms/FormField.js
Comment (albertparedandan) [reply ui/src/components/forms/FormField.js:1]: yes

**Review by albertparedandan — APPROVED** [review]

**Pull request merged**